File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/Week.java
package com.tickaroo.tikxml.annotationprocessing.textcontent;

import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Xml;
import java.util.List;

/**
 * @author Hannes Dorfmann
 */
@Xml
public class Week {

  @Element
  List<Day> days;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Week)) return false;

    Week week = (Week) o;

    return days != null ? days.equals(week.days) : week.days == null;
  }

  @Override public int hashCode() {
    return days != null ? days.hashCode() : 0;
  }
}

File: annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/Week.kt
package com.tickaroo.tikxml.annotationprocessing.textcontent

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml
class Week {

    @Element
    var days: List<Day>? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is Week) return false

        val week = o as Week?

        return if (days != null) days == week!!.days else week!!.days == null
    }

    override fun hashCode(): Int {
        return if (days != null) days!!.hashCode() else 0
    }
}
